agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java

```diff
@@ -1230,13 +1230,93 @@ public void validate() {
     }
   }
 
+  public enum RequestFilterType {
+    @JsonProperty("name-regex")
+    NAME_REGEX
+  }
+
+  public static class RequestFilter {
+    public RequestFilterType type;
+    public String value;
+  }
+
+  public static class RequestAggregationConfig {
+
+    // Threshold in ms over which a span will consider to be a breach
+    // Used by the breach ratio aggregation
+    public int thresholdMs = 5000;
+
+    // Minimum number of samples that must have been collected in order for the aggregation to
+    // produce data. Avoids volatile aggregation output on small sample sizes.
+    public int minimumSamples = 0;
+  }
+
+  public enum RequestAggregationType {
+    @JsonProperty("breach-ratio")
+    BREACH_RATIO
+  }
+
+  public static class RequestAggregation {
+    public RequestAggregationType type = RequestAggregationType.BREACH_RATIO;
+    public long windowSize = 60000; // in ms
+    public String evaluationFrequency = "every_sample";
+    public RequestAggregationConfig configuration = new RequestAggregationConfig();
+  }
+
+  public enum RequestTriggerThresholdType {
+    @JsonProperty("greater-than")
+    GREATER_THAN
+  }
+
+  public static class RequestTriggerThreshold {
+    public RequestTriggerThresholdType type = RequestTriggerThresholdType.GREATER_THAN;
+
+    // Threshold value applied to the output of the aggregation
+    // i.e :
+    //  - For the BreachRatio aggregation, 0.75 means this will trigger if 75% of sample breach the
+    // threshold.
+    //  - For a rolling average aggregation 0.75 will mean this will trigger if the average request
+    // processing time
+    //      breaches 0.75ms
+    public float value = 0.75f;
+  }
+
+  public enum RequestTriggerThrottlingType {
+    COOLDOWN
+  }
+
+  public static class RequestTriggerThrottling {
+    public RequestTriggerThrottlingType type = RequestTriggerThrottlingType.COOLDOWN;
+    public long value = 60000; // in ms
+  }
```

Comment:
An alternate could be RATE, i.e "No more than 2 profiles per hour". Could also do different types of COOLDOWN, i.e exponential backoff. If we were to encode different types of throttle, then it may be more accurate to call this FIXED_DURATION_COOLDOWN, that then leaves scope for other types of cooldown.

Reply:
I have updated it to FIXED_DURATION_COOLDOWN and aliased it to "fixed-duration-cooldown"